mantidproject/mantid
# Fix for VSI crash when using Pick Mode in the SplatterPlotView

Fixes #18736 

The pick mode option seems to have been causing a crash at least since Mantid3.6. This fixes PR fixes the crash. 

**To test:**

You will require an MDEventWorkspace and an associated PeaksWorkspace. If you are located at ISIS then you can use the workspaces from here: smb://olympic/babylon5/Scratch/Anton/SampleWS/Elliptical

The MDEventWorkspace is MDEvent_ellipPeaks_new
The peaks workspace is ellipPeaks_new

1. Load the MDEventWorkspace into the VSI and switch to the SplatterPlotView
2. Drag the PeaksWorkspace from MantidDock to the SplatterPlot render area
3. Follow these instructions below (from the VSI Wiki):

>A Pick Mode button is provided with this view. It is used in conjunction with a peaks workspace and the SCD Event Data Reduction (SEDR) interface. Pick mode can be entered by toggling the button and a peak can be picked by hovering over it with the mouse and pressing the p key. This can be repeated as necessary. To hook up the pick mode with the information display in the SEDR, do the following.

>    Open the SEDR (in the Interfaces menu) and switch to the Find Peaks tab
    Enter a workspace in the Peaks Workspace Name box
    Select the Use Existing Peaks Workspace radio button
    Click the Apply button
    Switch to the Point Info tab

>When a point is selected in pick mode, the SEDR retrieves the information about the peak closest to the point selected and displays it in the Point Info tab. If one has generated the viewed MDEventWorkspace and the corresponding PeaksWorkspace in the course of running the SEDR, the above steps are not necessary as they will already have been done. 

4. Click on the `Pick Mode` button in the VSI
   * Confirm that Mantid does not crash
5. Hover over a peak in the VSI and press p
   * Confirm that information appears in the SEDR GUI

#### Release notes

Please find them [here](https://github.com/mantidproject/mantid/pull/18756/commits/c42f36dc583d1dea5669b7de60b2fe4d015875a0)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
